schema/deploy/shipping/views.sql

```diff
@@ -1486,7 +1549,37 @@ create materialized view shipping.scan_encounters_v1 as
         social_precautions,
         no_mask,
         high_risk_feb2021,
-
+        overall_risk_jan2021,
+        covid_vax,
+        covid_doses,
+        vac_name_1,
+        vac_date,
+        vac_name_2,
+        vac_date_2,
+        case
+          when covid_vax = 'no' then 'not_vaccinated'
+          when covid_vax in (values('dont_know'), ('dont_say')) then 'unknown'
+          when covid_vax = 'yes' then
+            case
+              when vac_name_1 = 'johnson' and date_part('day', encountered::timestamp - vac_date::timestamp) >= 14 then 'fully_vaccinated'
+              when vac_name_1 in (values('pfizer'), ('moderna')) and vac_name_2 in (values('pfizer'), ('moderna')) and date_part('day', encountered::timestamp - vac_date_2::timestamp) >= 14 then 'fully_vaccinated'
```

Comment:
There's a funny edge case here, if a participant happened to mix manufacturers so one does of Pfizer and one dose of Moderna -- do we want to apply the same rules for calculating `vaccination_status` and `fully_vaccinated_date` in that case or do we need to differentiate that?

I'm not sure how these calculations will hold us as booster doses come into play, so I have mixed feelings about incorporating all of this logic into the shipping view. I think we'd need to get pretty specific guidance on how to keep these calculations up to date when SCAN starts collecting booster dose info from participants.

Reply:
Good point. I was basing this off of the original request that stated "fully_vaccinated is everyone who is 14 days after both mRNA doses or 14 days after J&J dose", which I interpreted as lumping the Pfizer and Moderna vaccines as mRNA doses.  So someone who got one of each would still be "fully_vaccinated" after their second dose, and the `vaccine_manufacturer` value would be "multiple" so would be possible to separate them out from those who got both doses from the same manufacturer.

I had the same thought about whether this logic belongs in the shipping view. It would be more flexible to have in Metabase, but I wasn't sure how comfortable they might be doing it there. But I'm be happy to drop it here if Metabase would be a better place for it.

Reply:
Some added context here: the request from Trevor was to add these to `shipping.scan_encounters_v2` which makes me think he's accessing the view directly. Since it was just adding columns to the view, I kept it as `scan_encounters_v1`.

Reply:
Confirmed with Trevor that he would prefer to keep these calculations here since there are multiple use cases (dashboards, manuscripts) and having consistency for `vaccination_status`, `vaccine_manufacturer` and `fully_vaccinated_date` would be useful. He and two others reviewed the updated query and agreed with how those values are being calculated.

Reply:
Great! Thanks for checking. Re: the booster dose situation, Trevor answered that too. Posting here for posterity/ when we update for that too.

> I'd think to keep fully_vaccinated_date and later add a vaccine_boosted_date for 2 weeks post booster